### DISCUSSION TITLE
How to specify ssh key for git repository?

### DISCUSSION THREAD

**Discussion by Sogl** [deployphp/deployer#3079] [Help needed] [answered]
I have ssh key in `~/.ssh/config`:
```
Host bitbucket.org
  IdentityFile ~/.ssh/my_key

```

This key is added in my repo and test connection works fine:
```
laradock@f56ba1192e13:/var/www/sogl$ ssh -T git@bitbucket.org
authenticated via ssh key.

You can use git to connect to Bitbucket. Shell access is disabled
```

BUT `dep deploy` command don't work:
```
Cloning into '/home/*****/public_html/*****/releases/28'...
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Connection to *server* closed.
```

I have lines with the same key for the server:
```php
host('*server*')
    ...
    ->configFile('~/.ssh/config')
    ->identityFile('~/.ssh/my_key')
    ...;
```

And I can't find any key option for repository:
```php
// Project repository
set('repository', 'git@bitbucket.org:Me/my-repo.git');
```

Any ideas?

Deployer 6.8.0

**Comment by antonmedv** [comment]
Your host doesn’t have same key. Two options: use agent forwarding or a separate key specially for your host (deploy keys).

**Comment by Sogl** [comment]
>use agent forwarding

According to documentation, it's [used by default](https://deployer.org/docs/6.x/advanced/deploy-and-git#agent-forwarding)
I added line `host(*server*)->forwardAgent()` but nothing changed.

>separate key specially for your host (deploy keys).

Do you mean creating a Bitbucket key on the remote/deploy server?

**Reply by antonmedv** [reply]
Yes, usually it is called deploy keys.

**Comment by Sogl** [comment]
But why agent forwarding isn't working?

**Reply by antonmedv** [reply]
Can’t tell.  Try to debug using -vvv

**Reply by antonmedv** [reply]
For me it’s working.

**Reply by Sogl** [reply]
How to check that's agent forwarding works? In which line of debug output?

**Comment by Sogl** [comment, marked as answer]
Agent forwarding finally works with:
```
eval `ssh-agent -s`
ssh-add ~/.ssh/my_key
```

Also +1 line `ForwardAgent yes` in `~/.ssh/config`, I don't know if it helped or not.

P.S. The last step is to find how to register it in Dockerfile
